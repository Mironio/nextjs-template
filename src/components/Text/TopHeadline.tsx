import { TextType } from '@/components/Text/TextType';

export default function TopHeadline({ classNames = '', children } : TextType) {
  return <h1 className={`${classNames} mb-6 text-[32px] font-bold text-pq-indigo`}>{children}</h1>;
}
