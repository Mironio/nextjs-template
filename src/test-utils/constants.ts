export const TestClassNames = 'vitest-1';
export const TestChildren = 'children';
