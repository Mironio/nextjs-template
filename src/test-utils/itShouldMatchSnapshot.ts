import { expect, it } from 'vitest';
import { JSX } from 'react';

export default function itShouldMatchSnapshot(jsx: JSX.Element) {
  it('should match snapshot', () => {
    expect(jsx).toMatchSnapshot();
  });
}
