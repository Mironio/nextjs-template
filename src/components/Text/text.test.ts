import itShouldMatchSnapshot from '@/test-utils/itShouldMatchSnapshot';
import Headline from '@/components/Text/HeadLine';
import { TestChildren, TestClassNames } from '@/test-utils/constants';
import TopHeadline from '@/components/Text/TopHeadline';

const HeadlineProps = { classNames: TestClassNames, children: TestChildren };
const TopHeadlineProps = { ...HeadlineProps };
itShouldMatchSnapshot(Headline(HeadlineProps));
itShouldMatchSnapshot(TopHeadline(TopHeadlineProps));
