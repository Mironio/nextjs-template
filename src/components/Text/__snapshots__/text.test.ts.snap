// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<h3
  className="vitest-1 text-2xl font-bold text-pq-indigo"
>
  children
</h3>
`;

exports[`should match snapshot 2`] = `
<h1
  className="vitest-1 mb-6 text-[32px] font-bold text-pq-indigo"
>
  children
</h1>
`;
